### ISSUE TITLE
Add GETDATE, NOW as alias to parser + allow CURRENT_TIMESTAMP with parenthesis

### ISSUE THREAD

**Issue by paulrutter** [agershun/alasql#1979]
We're in the process of using the AlaSQL parser to validate query syntax in our product.
The AlaSQL parser currently does not recognize the following aliases to `CURRENT_TIMESTAMP` (see `61date.js`).

- `NOW`
- `GETDATE`

Also, `CURRENT_TIMESTAMP` is recognized by the parser, while `CURRENT_TIMESTAMP()` is not.
Do note that all of these functions _do_ work in a query, the parser just cannot properly validate their use.

I'm going to supply a PR to get this fixed, which has one drawback: `now`, `current_timestamp` and `getdate` can no longer be used as column aliases, as they're now reserved keywords.

Let me know what you think @mathiasrw.

**Comment by paulrutter** [comment]
Ps. The same applies to other date functions like DAY() and YEAR(), but that would break far more queries (although it would be more consistent). I tried that locally a while ago. See https://github.com/AlaSQL/alasql/pull/1937#issuecomment-2198488165

With the proposed PR, at least all functions for getting the current date (+time) are consistently handled.

**Comment by mathiasrw** [comment]
Thanks so much for contributing! I'll merge this when we're ready to bump the major version. Since we're currently restructuring the code to be ES6-friendly, it might take a few months before we can release breaking changes.

**Comment by mathiasrw** [comment]
Is this a duplicate of https://github.com/AlaSQL/alasql/pull/1981?

**Comment by paulrutter** [comment]
> Is this a duplicate of #1981?

No, that seems entirely unrelated. Did you mean https://github.com/AlaSQL/alasql/pull/1982?

https://github.com/AlaSQL/alasql/pull/1982 indeed builds on this PR.
We could close https://github.com/AlaSQL/alasql/pull/1980 and just go for #1982 instead, as it adds far more date functions to the parser.
